Backend/app/rider-platform/rider-app/Main/src/Domain/Types/Booking/API.hs

```diff
@@ -138,6 +139,13 @@ makeBookingAPIEntity booking activeRide allRides fareBreakups mbExophone = do
               ..
             }
 
+mkSimulatedBookingAPIDetails :: SSRL.SearchReqLocation -> HighPrecMeters -> BookingAPIDetails
+mkSimulatedBookingAPIDetails toLocation distance = OneWayAPIDetails $ do
```

Comment:
all this additional logic can be within `SharedLogic.SimulatedFlow`, we don;t really need to use it in main flow in any way. You may even copy main project structure within `SharedLogic.SimulatedFlow`, maybe with Storage folder even. So things from it won't appear in main code in any way (except these interceptWhenSimulated calls)